### PR TITLE
App now turns on with R, runs as background process on all pages

### PR DESCRIPTION
**Pull request by benfrancis314**

### CODE REVIEW

File: extension/background.js [hunk 1]
```diff
@@ -1,82 +1,82 @@
...
+//             // Send a message which will get handled by content_script.js
+//             // This ensures that the same isolated world which has the first and only 
+//             // script execution can handle the event.
+//             // The alternative approach of executeScripts and rerunning content_script.js
+//             // does not work because it will execute in a different isolated world,
+//             // and will have lost access to the previous javascript vars and event handlers.
+//             // See https://stackoverflow.com/a/8916706/4143394
+//             chrome.tabs.sendMessage(tab.id, {command: "toggleUI"}, function(response) {});
+//         });
+// });
```
Comment (benfrancis314) [inline extension/background.js]: Left in case we want to turn it back; I can also just delete this file

File: extension/content_script.js [hunk 2]
```diff
@@ -402,7 +402,10 @@ function toggleKeywordSettings() {
 
 function setupKeyListeners() {
 	let wpmDisplay = $("#speedContainer");
-	jdoc.on("keydown", function(evt) {
+	/* This creates an event namespace so that only these events
+		are removed when off() is called in removeUI. 
+		See: https://api.jquery.com/on/ , section "Event names and namespaces" */
+	jdoc.on("keydown.running", function(evt) {
```
Comment (benfrancis314) [inline extension/content_script.js:408]: Btw can definitely change the name of this namespace

File: extension/background.js (hunk 1 above)
Comment (sjonany) [reply extension/background.js]: I would just delete it. One less file + package.json line.

File: extension/content_script.js (hunk 2 above)
Comment (sjonany) [reply extension/content_script.js:408]: very nice usage of the namespace feature!

File: extension/content_script.js [hunk 3]
```diff
@@ -518,6 +522,7 @@ function oneTimeSetup() {
 			}
 		);
 	});
+	cb();
```
Comment (sjonany) [inline extension/content_script.js]: no ben we talked about this

-- jk or maybe you meant cb() _after settings is loaded?
another option is to just put setupKeyListenerForOnOff() right after settings.getSpeed

File: extension/content_script.js [hunk 4]
```diff
@@ -557,5 +561,23 @@ function toggleExtensionVisibility() {
...
+		if (!document.hasFocus()) {
+		  return true;
+		}
+		// Make sure the user isn't trying to type anything
+		// If there are exceptions to this it should hopefully come up during testing
+		// There probably will be exceptions, so the key is WHAT are the exceptions
+		let focuses = $(":focus");
+		if (focuses.is("input") || focuses.is("form") || focuses.is("textarea")) { console.log("input"); return }
+
+		if (evt.code == 'KeyR') {
```
Comment (sjonany) [inline extension/content_script.js:568]: TODO:  on clickup to update instruction

File: extension/content_script.js [hunk 5]
```diff
@@ -557,5 +561,23 @@ function toggleExtensionVisibility() {
...
 
-oneTimeSetup();
+function setupKeyListenerForOnOff() {
+	jdoc.on("keydown", function(evt) {
+		if (!document.hasFocus()) {
+		  return true;
+		}
+		// Make sure the user isn't trying to type anything
+		// If there are exceptions to this it should hopefully come up during testing
+		// There probably will be exceptions, so the key is WHAT are the exceptions
```
Comment (sjonany) [inline extension/content_script.js]: hm can u leave TODO: google in stack overflow?
these things usually have a nice jquery / someone already cover all the edgecases on stackoverflow

or if you already have a stackoverflow link pls link it here

**Review by sjonany — CHANGES_REQUESTED** [review]

File: extension/content_script.js (hunk 3 above)
Comment (benfrancis314) [reply extension/content_script.js]: First option is what was meant; if "R" is clicked before this is done setting up it causes problems

**Review by sjonany — APPROVED** [review]
LGTM!
Very clean solution!

**Pull request merged**